File: complicaus_2/maximum_distance.py
# Source: Codeforces Gym
# URL: https://codeforces.com/gym/102951/problem/A
# Solution by Kenny Jesús Flores Huamán

n = int(input())
x_coords = list(map(int, input().split()))
y_coords = list(map(int, input().split()))

def solve(n, x_coords, y_coords):
    res = 0
    for i in range(n):
        for j in range(i+1, n):
            d = (x_coords[j] - x_coords[i])**2 + (y_coords[j]- y_coords[i])**2
            res = max(res, d)
    
    return res


print(solve(n, x_coords, y_coords))

File: complicaus_2/maximum_distance.hs
-- Source: Codeforces Gym
-- URL: https://codeforces.com/gym/102951/problem/A
-- Solution by Alejandro Domínguez Muñoz

{-# LANGUAGE TupleSections #-}

import Data.List

solve = maximum . fmap (uncurry pythagoras) . (=<<) pairsWithFirst . tails

pairsWithFirst (x:xs) = fmap (x,) xs
pairsWithFirst _      = []

pythagoras (x, y) (x', y') = ((x - x') ^ 2) + ((y - y') ^ 2)

parse :: String -> Maybe [(Int, Int)]
parse = fmap (uncurry zip) . listToTuple . fmap (fmap read . words) . drop 1 . lines

listToTuple :: [[a]] -> Maybe ([a], [a])
listToTuple [x, y] = Just (x, y)
listToTuple _      = Nothing

run :: String -> Maybe Int
run = fmap solve . parse

main :: IO ()
main = interact (maybe "" show . run)
